request.go

```diff
@@ -851,6 +853,18 @@ func (r *Request) WithMultipart() *Request {
 	return r
 }
 
+// WithRetry allows to retry an HTTP request until a max retry value.
+// It also accepts a retry delay that is performed before to retry on failure.
+func (r *Request) WithRetry(maxRetry int, retryDelay time.Duration) *Request {
```

Comment:
Is the retryDelay parameter useful in practice? (In your case or in general). Maybe instead remove it and use exponential back-off, to make the API a bit simpler?